pkg/pgmodel/upgrade_tests/upgrade_test.go

```diff
@@ -292,6 +204,114 @@ func TestUpgradeFromPrev(t *testing.T) {
 	}
 }
 
+func withDBAndConnector(t testing.TB, DBName string, connectorImage string,
+	f func(dbContainer testcontainers.Container, dbTmpDir string, connectorHost string, connectorPort nat.Port)) {
+	var err error
+	ctx := context.Background()
+
+	tmpDir, err := testhelpers.TempDir("update_test_out")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	dbContainer, err := testhelpers.StartPGContainerWithImage(ctx, cleanImage, tmpDir, *printLogs)
+	if err != nil {
+		t.Fatal("Error setting up container", err)
+	}
+
+	defer func() {
+		if *printLogs {
+			_ = dbContainer.StopLogProducer()
+		}
+
+		_ = dbContainer.Terminate(ctx)
+	}()
+
+	db, err := testhelpers.DbSetup(*testDatabase, testhelpers.Superuser)
+	if err != nil {
+		t.Fatal(err)
+		return
+	}
+	db.Close()
+
+	connector, err := testhelpers.StartConnectorWithImage(context.Background(), connectorImage, *printLogs, *testDatabase)
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+	defer func() {
+		if *printLogs {
+			_ = connector.StopLogProducer()
+		}
+		_ = connector.Terminate(context.Background())
+	}()
+
+	connectorHost, err := connector.Host(ctx)
+	if err != nil {
+		t.Fatal(err)
+		return
+	}
+
+	connectorPort, err := connector.MappedPort(ctx, testhelpers.ConnectorPort)
+	if err != nil {
+		t.Fatal(err)
+		return
+	}
+	f(dbContainer, tmpDir, connectorHost, connectorPort)
+}
+
+func withNewDB(t testing.TB, initialVersion string, DBName string,
```

Comment:
```suggestion
func withDBInstalledAtLatestVersion(t testing.TB, DBName string,
```
The version should not be passed in. This function should look up version.Version